linetools/spectra/xspectrum1d.py

```diff
@@ -526,3 +531,85 @@ def write_to_fits(self, outfil, clobber=True, add_wave=False):
         # Write
         hdu.writeto(outfil, clobber=clobber)
         print('Wrote spectrum to {:s}'.format(outfil))
+
+
+    def fit_continuum(self, knots=None, edges=None, wlim=None, dw=10.,
+                      kind=None, **kwargs):
+        """ Interactively fit a continuum.
+
+        Parameters
+        ----------
+        spec : Spectrum1D
+        wlim : (float, float), optional
+          Start and end wavelengths for fitting the continuum. Default is
+          None, which fits the entire spectrum.
+        knots: list of (x, y) pairs, optional
+          A list of spline knots to use for the continuum.
+        edges: list of floats, optional
+          A list of edges defining wavelength chunks. Spline knots
+          will be placed at the centre of these chunks.
+        dw : float, default 10.0
+          The approximate distance between spline knots in
+          Angstroms.
+        kind : {'QSO', None}, default None
+          If given, generate spline knots using
+          linetools.analysis.continuum.find_continuum.
+
+        Updates
+        -------
+        spec.co with the new continuum
+        spec.meta['contpoints'] with the knots defining the continuum
+
+        Use linetools.analysis.interp.AkimaSpline to regenerate the
+        continuum from the the knots.
+        """
+
+        wa = self.dispersion.value
+
+        if wlim is None:
+            wmin, wmax = wa[0], wa[-1]
+        else:
+            wmin, wmax = wlim
+            if wmax < wmin:
+                wmin, wmax = wmax, wmin
+
+        if kind is not None:
+            _, knots = find_continuum(self, kind=kind, **kwargs)
+            # only keep knots between wmin and wmax
+            knots = [[x,y] for (x,y) in knots if wmin < x < wmax]
+        else:
+            if edges is None:
+                nchunks = max(3, (wmax - wmin) / float(dw))
+                edges = np.linspace(wmin, wmax, nchunks + 1)
+    
+        if knots is None:
+            knots, indices, masked = prepare_knots(
+                wa, self.flux.value, self.uncertainty.array, edges)
+        else:
+            knots = [list(k) for k in knots]
+    
+        co = (self.co if hasattr(self, 'co') else None)
```

Comment:
This raises an error in the continuum fitting, when self.co = None (i.e. self.co is defined but its value is None). 

Solutions:
- Initialize self.co within self.fit_continuum()
- Initialize self.co at the XSpectrum1D() initialization stage

My preference is for the second option, using something like 

self.co = self.flux * float(X)

so it is the same type and units as self.flux and has the same units. float(X) could be 0. to emphasize that no continuum estimation exists, but it could also be a fully automatic rough estimation.